awslabs/aws-lambda-powertools-typescript
# refactor(docs): apply eslint rules to code snippets

## Description of your changes


1. introduces : `root: true` in eslint configuration file in root of repository which manages the workspace. 
2. applies existing eslint rules defined in eslint configuration file in root of repository
3. fixes code snippets in docs/snippets lintinting issues which are related to styling:  `semi`,  `quotes`, `indent`    
4.  introduces overrides section in eslint configuration file in root of repository, so that we can discuss explicit about typescript linting issues for code snippets: `no-explicit-any`, `no-unused-vars`, `explicit-member-accessibility`, `no-var-requires`  ,
5. overrides section in eslint configuration file in root of repository, has no impact or side effects to the overall code base of this repository. 
6. missing dependency of `@aws-sdk/client-s3` defined as dev. dependency in package.json of docs/snippets 






TODOs:
options / trade-offs, typescript linting issues for code snippets: 
- `no-explicit-any`, `no-unused-vars`, `explicit-member-accessibility`, `no-var-requires`

### How to verify this change



1. Verify that root eslint configuration file is used by running: `npm run -w docs/snippets lint -- --debug`
should yield
```
...
 eslintrc:config-array-factory Loading JS config file: /aws-lambda-powertools-typescript/.eslintrc.js +1ms
...
eslintrc:cascading-config-array-factory Stop traversing because of 'root:true'. +806ms

```
2. Run: `npm run -w docs/snippets lint`




### Related issues, RFCs



**Issue number:** #1252 #729

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
